File: src/structural/flyweight/python/delivery_context.py
from classes.factories.delivery_factory import DeliveryFactory
from classes.interfaces.delivery_location_data import DeliveryLocationData


def deliveryContext(
    factory: DeliveryFactory,
    name: str,
    number: str,
    street: str,
    city: str,
) -> None:
    location = factory.create_location(DeliveryLocationData(city, street))
    location.deliver(name, number)

File: src/structural/flyweight/typescript/deliveryContext.ts
import { DeliveryFactory } from './classes/factories/DeliveryFactory';

export const deliveryContext = (
  factory: DeliveryFactory,
  name: string,
  number: string,
  street: string,
  city: string,
): void => {
  const location = factory.createLocation({ city, street });
  location.deliver(name, number);
};
